### PR TITLE
Add pFilterIterable

### PR DESCRIPTION
**Pull request by noamokman**
Hey!

I wanted to add `pFilterIterable` to be able to filter async iterables.

I'm using `pMapIterable` and the code is pretty simple.

Added tests and types,
Let me know if I missed anything.

Thanks!

### CODE REVIEW

**Comment by noamokman** [comment]
@sindresorhus can you take a look? 🙏🏻

File: index.d.ts [hunk 1]
```diff
@@ -38,4 +38,44 @@ export default function pFilter<ValueType>(
...
+const filterer = async place => {
+	const weather = await getWeather(place);
+	return weather.temperature > 30;
+};
+
+const result = await pFilterIterable(places, filterer);
+
+console.log(result);
+//=> ['Bangkok, Thailand']
+```
```
Comment (sindresorhus) [inline index.d.ts:74]: The example is not correct. And the example should also be in the readme under the `pFilterIterable` section.

File: readme.md [hunk 2]
```diff
@@ -64,6 +64,34 @@ Minimum: `1`
 
 The number of concurrently pending promises returned by `filterer`.
 
+### pFilterIterable(iterable, filterer, options?)
+
+Returns an async iterable that iterates over the promises in `iterable` and ones returned from `filterer` concurrently, calling `filterer` for each element.
```
Comment (sindresorhus) [inline readme.md:71]: index.d.ts and the readme should be in sync as much as possible.

**Review by sindresorhus — CHANGES_REQUESTED** [review]

File: index.d.ts (hunk 1 above)
Comment (noamokman) [reply index.d.ts:74]: fixed 👍🏻

**Comment by sindresorhus** [comment]
Thanks :)

**Pull request merged**